scripts/serve_header/README.md

```diff
@@ -0,0 +1,81 @@
+serve_header.py
+===============
+
+Serves the `single_include/nlohmann/json.hpp` header file over HTTP(S).
+
+The header file is automatically amalgamated on demand.
+
+## Prerequisites
+
+> **Note:** All file locations are relative to the project root folder and all commands are assumed to be executed from the same root folder.
+
+1. Make sure these Python packages are installed.
+    ```
+    PyYAML
+    watchdog
+    ```
+
+2. To serve the header over HTTPS (which is required by Compiler Explorer at this time), a certificate is needed.
+   The recommended method for creating a locally-trusted certificate is to use [`mkcert`](https://github.com/FiloSottile/mkcert).
+   - Install the `mkcert` certificate authority into your trust store(s):
+     ```
+     $ mkcert -install
+     ```
+   - Create a certificate for `localhost`:
+     ```
+     $ mkcert localhost
+     ```
+
+## Usage
+
+`serve_header.py` has a builtin default configuration that will serve the header from the project folder it is launched from.
+It expects to find the certificate `localhost.pem` and the private key `localhost-key.pem` in the projects root directory.
+
+To start serving the `json.hpp` header file at `https://localhost:8443/json.hpp`, run this command:
+```
+$ make serve_header
+```
+
+Open [Compiler Explorer](https://godbolt.org/) and try it out:
+```cpp
+#include <https://localhost:8443/json.hpp>
+using namespace nlohmann;
+
+#include <iostream>
+
+int main() {
+    // these macros are dynamically injected into the header file
+    std::cout << JSON_BUILD_TIME << " (" << JSON_BUILD_COUNT << ")\n";
+
+    return 0;
+}
+```
+
+> `serve_header.py` dynamically injects the macros `JSON_BUILD_COUNT` and `JSON_BUILD_TIME` into the served header. By comparing build count or time output from the compiled program with the output from `serve_header.py`, one can be reasonably sure the compiled code uses the expected revision of the header file.
+
+## Configuration
+
+`serve_header.py` will try to read a configuration file `serve_header.yml` and will fallback on builtin defaults, if the file cannot be read.
+An annotated example configuration can be found in `scripts/serve_header/serve_header.yml.example`.
+
+## Serving `json.hpp` from multiple working trees
```

Comment:
Does the approach also work when working without git worktree?

Reply:
Whatever you point the `root` configuration option to, will be scanned for project directories (defined somewhat arbitrarily by the existence of `Makefile`, `include/nlohmann/`, and `single_include/nlohmann/`).

Reply:
Then I would add such a note to the README. I have not tried it yet, but it sounded to me that you expect a different structure.

Reply:
You can preview the changes in the PR description until I push an update later.